pumps/graph_sql.go

```diff
@@ -0,0 +1,174 @@
+package pumps
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	gorm_logger "gorm.io/gorm/logger"
+)
+
+const (
+	GraphSQLPrefix = "GraphSQL-Pump"
+	GraphSQLTable  = "tyk_analytics_graph"
+)
+
+type GraphSQLPump struct {
+	db        *gorm.DB
+	SQLConf   SQLConf
+	tableName string
+	CommonPumpConfig
+}
+
+func (g *GraphSQLPump) GetName() string {
+	return "Graph SQL Pump"
+}
+
+func (g *GraphSQLPump) New() Pump {
+	return &GraphSQLPump{}
+}
+
+func (g *GraphSQLPump) Init(conf interface{}) error {
+	g.log = log.WithField("prefix", GraphSQLPrefix)
+
+	if err := mapstructure.Decode(conf, &g.SQLConf); err != nil {
```

Comment:
@kofoworola we are missing a way to set the things thru env variables. Please check how SQL pump does it
https://github.com/TykTechnologies/tyk-pump/blob/master/pumps/sql.go#L136
https://github.com/TykTechnologies/tyk-pump/blob/master/pumps/sql.go#L105